src/main/java/com/sleekbyte/tailor/Tailor.java

```diff
@@ -193,7 +204,9 @@ private void exitWithNoSourceFilesError() {
                 swiftParser.addErrorListener(new ErrorListener());
             }
             tree = Optional.of(swiftParser.topLevel());
+            unlink(swiftParser);
         } catch (CLIArgumentParserException e) {
+            unlink(swiftParser);
             handleCLIException(e);
         }
```

Comment:
The changes introduced here could be replaced by the following after the `catch` block:

```java
finally {
    swiftParser.getInterpreter().clearDFA();
}
```